vivisect/parsers/elf.py

```diff
@@ -323,18 +333,59 @@ def loadElfIntoWorkspace(vw, elf, filename=None):
 
     for s in elf.getSymbols():
         sva = s.st_value
+
+        # if the symbol has a value of 0, it is likely a relocation point which gets updated
+        sname = normName(s.name)
+        if sva == 0:
+            for reloc in relocs:
+                rname = normName(reloc.name)
+                if rname == sname:
+                    sva = reloc.r_offset
+                    break
+
+        origname = sname
+        demangledname = demangle(sname)
+
         if addbase: sva += baseaddr
-        if vw.isValidPointer(sva) and len(s.name):
+        if vw.isValidPointer(sva) and len(demangledname):
             try:
-                vw.makeName(sva, s.name, filelocal=True)
+                vw.makeName(sva, demangledname, filelocal=True)
             except Exception, e:
                 print "WARNING:",e
 
-    if vw.isValidPointer(elf.e_entry):
-        vw.addExport(elf.e_entry, EXP_FUNCTION, '__entry', fname)
-        vw.addEntryPoint(elf.e_entry)
+    if addbase:
+        eentry = baseaddr + elf.e_entry
+    else:
+        eentry = elf.e_entry
+
+    if vw.isValidPointer(eentry):
+        vw.addExport(eentry, EXP_FUNCTION, '__entry', fname)
+        vw.addEntryPoint(eentry)
         
     if vw.isValidPointer(baseaddr):
         vw.makeStructure(baseaddr, "elf.Elf32")
 
     return fname
+
+def normName(name):
+    '''
+    Normalize symbol names.  ie. drop the @@GOBBLEDEGOOK from the end
+    '''
+    atidx = name.find('@@')
+    if atidx > -1:
+        name = name[:atidx]
+    return name
+
+def demangle(name):
+    '''
+    Translate C++ mangled name back into the verbose C++ symbol name (with helpful type info)
+    '''
+    name = normName(name)
+
+    try:
+        import cxxfilt
+        name = cxxfilt.demangle(name)
+    except:
+        pass
```

Comment:
except: pass is the devil.  is this to catch a cxxfilt.demangle exception or the import error?

Reply:
it is indeed the devil.  perhaps a logger.debug() call.  
this is for either.  
if cxxfilt isn't there, it skips it
if the name doesn't demangle correctly, it raises and exception.

Reply:
fixed.